### PR TITLE
feat: activates logic to explicitly cancel an execution

### PR DESCRIPTION
**Pull request by sfishel18**
removes the protocol version check that was gating the logic to send a cancel message to the server when the execution is cancelled by the caller.

this exposed a leak in the listeners being added to abort controllers, which is fixed by proper listener teardown logic

bonus: adds better messaging to test that will incorrectly fail if the WHEROBOTS_API_KEY env var is set in the terminal session where tests are run

### CODE REVIEW

**Review by mpetazzoni — COMMENTED** [review]
LGTM, although @ClayMav should review. It doesn't look like you're relying on the server responding to the cancellation request to control/trigger the cleanup, so this makes the protocol handling logic here a bit different than in the other clients, but that's fine.

**Comment by sfishel18** [comment]
> It doesn't look like you're relying on the server responding to the cancellation request to control/trigger the cleanup, so this makes the protocol handling logic here a bit different than in the other clients, but that's fine.

right, we already had a cancellation interface exposed to the consumer that would do the listener cleanup, so i figured i'd keep it simple and just add the sending of the cancellation message to that process in a "fire and forget" way. shouldn't be too hard to add logic to wait for the response if we need it later

**Review by mpetazzoni — APPROVED** [review]

File: examples/connectionWithCancellation.ts
Comment (ClayMav) [inline examples/connectionWithCancellation.ts:16]: Can we support the new runtime names?

**Review by ClayMav — APPROVED** [review]

File: examples/connectionWithCancellation.ts
Comment (sfishel18) [reply examples/connectionWithCancellation.ts:16]: good call, will fix that in a follow-on pr

**Pull request merged**
